function.go

```diff
@@ -0,0 +1,126 @@
+package currency
+
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
+// validateISOCodeAlpha : returns a formatted ISO alpha code or an error if the ISO is not found
+func validateISOCodeAlpha(alpha string) (string, error) {
+	alpha = strings.ToUpper(alpha)
+	for key := range CurrencyList {
+		if key == alpha {
+			return alpha, nil
+		}
+	}
+	return "", errorInvalidISO
+}
+
+// validateISOCodeNumeric : returns a formatted ISO numeric code or an error if the ISO is not found
+func validateISOCodeNumeric(num string) (string, error) {
+	for _, value := range CurrencyList {
+		if value.Numeric == num {
+			return value.Numeric, nil
+		}
+	}
+	return "", errorInvalidISO
+}
+
+// getISOFromAlpha : returns the Currency struct for that ISO code alpha
+func getISOFromAlpha(alpha string) (Currency, error) {
+	alpha, err := validateISOCodeAlpha(alpha)
+	if err != nil {
+		return Currency{}, err
+	}
+	output := CurrencyList[alpha]
+	return output, nil
+}
+
+// getAlphaFromISOCodeNumeric : returns a formatted ISO alpha code from the ISO numeric counterpart
+func getAlphaFromISOCodeNumeric(num string) (string, error) {
+	for _, value := range CurrencyList {
+		if value.Numeric == num {
+			return value.Alpha, nil
+		}
+	}
+	return "", errorInvalidISO
+}
+
+// convertToString : returns the uint number as a string
+func convertToString(num uint, exp int) string {
```

Comment:
Might want to be more specific. This function does more than just convert a uint to a string.